src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/utilities/PointerFinder.java

```diff
@@ -84,43 +84,9 @@ public static PointerLocation find(Address a) {
 
     // Check if address is in the java heap.
     CollectedHeap heap = VM.getVM().getUniverse().heap();
-    if (heap instanceof GenCollectedHeap) {
-      GenCollectedHeap genheap = (GenCollectedHeap) heap;
-      if (genheap.isIn(a)) {
-        loc.heap = heap;
-        for (int i = 0; i < genheap.nGens(); i++) {
-          Generation g = genheap.getGen(i);
-          if (g.isIn(a)) {
-            loc.gen = g;
-            break;
-          }
-        }
-
-        if (Assert.ASSERTS_ENABLED) {
-          Assert.that(loc.gen != null, "Should have found this in a generation");
-        }
-
-        if (VM.getVM().getUseTLAB()) {
-          // Try to find thread containing it
-          for (int i = 0; i < threads.getNumberOfThreads(); i++) {
-            JavaThread t = threads.getJavaThreadAt(i);
-            ThreadLocalAllocBuffer tlab = t.tlab();
-            if (tlab.contains(a)) {
-              loc.inTLAB = true;
-              loc.tlabThread = t;
-              loc.tlab = tlab;
-              break;
-            }
-          }
-        }
```

Comment:
I don't know if I should restore these codes for TLAB pointer identification.

Reply:
Why not? From my understanding of the changes, all that should be needed for SA is to rename GenCollectedHeap to SerialHeap and move (most of) the contents of GenCollectedHeap.java to SerialHeap.java. I would use that as a starting point and see if everything works ok.